File: leetcode/1.ts
function twoSum(nums: number[], target: number): number[] {
  let hmap = new Map<number, number>();
  for(let i = 0; i < nums.length; i++) {
      let diff = target - nums[i];
      if(diff in hmap && hmap[diff] != i) {
          return [hmap[diff], i];
      }
      hmap[nums[i]] = i;
  }
  return[]
};

File: leetcode/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        vector<int> ans(2);
        for(auto i = 0; i<nums.size(); i++) {
            int complement = target - nums[i];
            if(mp.find(complement)!= mp.end()) {
                ans[0] = mp[complement];
                ans[1] = i;
                return ans;
            }
            mp[nums[i]] = i;
        }
        return {};
    }
};
